src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/AndroidCertificatePal.cs

```diff
@@ -25,6 +26,13 @@ public static ICertificatePal FromHandle(IntPtr handle)
             if (handle == IntPtr.Zero)
                 throw new ArgumentException(SR.Arg_InvalidHandle, nameof(handle));
 
+            if (Interop.AndroidCrypto.IsKeyStorePrivateKeyEntry(handle))
+            {
+                var newPrivateKeyEntryHandle = new Interop.JObjectLifetime.SafeJObjectHandle();
+                Marshal.InitHandle(newPrivateKeyEntryHandle, Interop.JObjectLifetime.NewGlobalReference(handle));
```

Comment:
I've seen these lines paired together a couple of times. A nonblocking suggestion would be to add a new static method to `SafeJObjectHandle` that creates a global reference from an `IntPtr`

```C#
internal partial class SafeJObjectHandle
{
    internal static SafeJObjectHandle CreateGlobalReferenceFromHandle(IntPtr handle)
    {
        var jObjectHandle = new SafeJObjectHandle();
        Marshal.InitHandle(jObjectHandle, Interop.JObjectLifetime.NewGlobalReference(handle));
        return jObjectHandle;
    }    
}
```